airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/InitialSnapshotHandler.java

```diff
@@ -90,16 +91,21 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIterators(
               // if nothing was found, return a new BsonDocument
               .orElseGet(BsonDocument::new);
 
-          final var cursor = collection.find()
+          final var cursor = isEnforceSchema ? collection.find()
```

Comment:
nit : can we have an if/else for clarity and comments on how the queries differ

Reply:
Added comment but left the `? :` operator